src/SprykerEco/Zed/AvalaraTax/Business/Expander/WarehouseExpander.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AvalaraTax\Business\Expander;
+
+use Generated\Shared\Transfer\ItemTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Zed\AvalaraTax\Business\Reader\StockProductReaderInterface;
+
+class WarehouseExpander implements WarehouseExpanderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\AvalaraTax\Business\Reader\StockProductReaderInterface
+     */
+    protected $stockProductReader;
+
+    /**
+     * @param \SprykerEco\Zed\AvalaraTax\Business\Reader\StockProductReaderInterface $stockProductReader
+     */
+    public function __construct(StockProductReaderInterface $stockProductReader)
+    {
+        $this->stockProductReader = $stockProductReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function expandQuoteItemsWithWarehouse(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        if (!$quoteTransfer->getItems()->count()) {
+            return $quoteTransfer;
+        }
+
+        $storeName = $quoteTransfer->getStoreOrFail()->getNameOrFail();
+        $indexedStockProductTransfers = $this->stockProductReader->getStockProductsIndexedByProductConcreteSku(
+            $this->extractItemSkus($quoteTransfer),
+            $storeName
+        );
+
+        $indexedStockTransfers = $this->stockProductReader->getStocksIndexedByName(
+            $this->extractStockNames($indexedStockProductTransfers),
+            $storeName
+        );
+
+        foreach ($quoteTransfer->getItems() as $itemTransfer) {
+            $this->expandQuoteItemWithWarehouse($itemTransfer, $indexedStockProductTransfers, $indexedStockTransfers);
```

Comment:
$itemTransfer = $this->expandQuoteItemWithWarehouse($itemTransfer, $indexedStockProductTransfers, $indexedStockTransfers);